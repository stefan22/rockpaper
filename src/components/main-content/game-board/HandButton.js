import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const HandButton = (props) => {

  return (
    <div className='col-4'>
      <Paper className='game-hand'>
        <img
          className='rps-button'
          onClick={() => props.onPlayerHandPick(props.choice)}
          src={props.imgsrc} alt={props.imgsrc}
        />
        <Typography className='game-name' gutterBottom variant="h4" component="h4">
          {props.hand}
        </Typography>

      </Paper>
    </div>
  );

};


export default HandButton;




