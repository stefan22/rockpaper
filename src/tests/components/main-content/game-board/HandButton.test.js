import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import HandButton from  '../../../../components/main-content/game-board/HandButton';

describe('HandButton component', () => {
  it('should render HandButton correctly', () => {
    const wrapper = shallow(<HandButton />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should have a class of "game-hand"', () => {
    const wrapper = shallow(<HandButton />);
    expect(wrapper.find('.game-hand').length).toBe(1);
  });

});


