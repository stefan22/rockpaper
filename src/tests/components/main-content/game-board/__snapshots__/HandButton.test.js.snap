// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandButton component should render HandButton correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HandButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Paper)
        className="game-hand"
      >
        <img
          className="rps-button"
          onClick={[Function]}
        />
        <WithStyles(Typography)
          className="game-name"
          component="h4"
          gutterBottom={true}
          variant="h4"
        />
      </WithStyles(Paper)>,
      "className": "col-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            className="rps-button"
            onClick={[Function]}
          />,
          <WithStyles(Typography)
            className="game-name"
            component="h4"
            gutterBottom={true}
            variant="h4"
          />,
        ],
        "className": "game-hand",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "rps-button",
            "onClick": [Function],
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": "game-name",
            "component": "h4",
            "gutterBottom": true,
            "variant": "h4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Paper)
          className="game-hand"
        >
          <img
            className="rps-button"
            onClick={[Function]}
          />
          <WithStyles(Typography)
            className="game-name"
            component="h4"
            gutterBottom={true}
            variant="h4"
          />
        </WithStyles(Paper)>,
        "className": "col-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <img
              className="rps-button"
              onClick={[Function]}
            />,
            <WithStyles(Typography)
              className="game-name"
              component="h4"
              gutterBottom={true}
              variant="h4"
            />,
          ],
          "className": "game-hand",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "className": "rps-button",
              "onClick": [Function],
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "game-name",
              "component": "h4",
              "gutterBottom": true,
              "variant": "h4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
